### PR TITLE
completed plp page along with functionalities and intigreted with nav…

### PR DESCRIPTION
**Pull request by swathiiN**
Screenshot (44)
Screenshot (45)

### CODE REVIEW

File: src/libs/ui/src/molecules/PlpAccordians/index.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+  const handleSortChange = (sortOption: string) => {
+    setSelectedSort(sortOption);
+    onSortChange(sortOption);
+  };
+
+  // Sort options
+  const sortOptions = ['Price: Low to High', 'Price: High to Low', 'New Arrivals'];
+
+  // Define checkbox accordions
+  const checkboxAccordions = [
```
Comment (Vishnu9901) [inline src/libs/ui/src/molecules/PlpAccordians/index.tsx]: Please remove this checkboxAccordions variable from here, add to top of the file

File: src/libs/ui/src/molecules/PlpAccordians/index.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  };
+
+  // Handle sort option change
+  const handleSortChange = (sortOption: string) => {
+    setSelectedSort(sortOption);
+    onSortChange(sortOption);
+  };
+
+  // Sort options
+  const sortOptions = ['Price: Low to High', 'Price: High to Low', 'New Arrivals'];
```
Comment (Vishnu9901) [inline src/libs/ui/src/molecules/PlpAccordians/index.tsx]: Please remove above variable here

File: src/libs/ui/src/molecules/PlpAccordians/index.tsx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+  const [showAllCategories, setShowAllCategories] = useState(false);
+  const [selectedProductCategory, setSelectedProductCategory] = useState<string | null>(null);
+  const [checkedFilters, setCheckedFilters] = useState<{ [key: string]: boolean }>({});
+  const [isSmallScreen, setIsSmallScreen] = useState<boolean>(false);
+  const [selectedSort, setSelectedSort] = useState<string>(''); // State for sort option
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false); // State for modal visibility
+
+  // Responsive check for small screens
+  useEffect(() => {
+    const handleResize = () => {
```
Comment (Vishnu9901) [inline src/libs/ui/src/molecules/PlpAccordians/index.tsx]: move useEffect last of the page

File: src/libs/ui/src/molecules/PlpAccordians/index.tsx [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+              ulClassName={ulClassName}
+              liClassName={liClassName}
+            />
+          </div>
+        </Modal>
+      ) : (
+        <div>
+          {/* Desktop View - Filter Options */}
+          {/* Product Category Accordion */}
+          <LabelAccordion
```
Comment (Vishnu9901) [inline src/libs/ui/src/molecules/PlpAccordians/index.tsx]: below filter items should be dynamic

File: src/libs/ui/src/organisms/Header/index.tsx [hunk 5]
```diff
@@ -108,32 +109,93 @@ const Header: React.FC<headerProps> = ({ modalSetToggle, handleRegisterClick })
...
+      const encodedTitle = title.replace(/[\s&]+/g, '-');
+      if (selectNavLink === appSetting[3].title) {
+        navigate(`${selectNavLink}/${encodedTitle}`);
+      }
+
+      // const searchParams = new URLSearchParams(location.search); // Create a new instance of URLSearchParams
+
+      // Create an array of navigation links with their corresponding query parameters
+      const navLinkMap = [
+        { title: appSetting[0].title, category: encodedTitle },
```
Comment (Vishnu9901) [inline src/libs/ui/src/organisms/Header/index.tsx]: if appSetting[0] is undefined then we will get error, please check this condition

File: src/libs/ui/src/organisms/Header/index.tsx [hunk 6]
```diff
@@ -108,32 +109,93 @@ const Header: React.FC<headerProps> = ({ modalSetToggle, handleRegisterClick })
...
-      navigate(`${selectNavLink}/${encodedTitle}`);
+      const encodedTitle = title.replace(/[\s&]+/g, '-');
+      if (selectNavLink === appSetting[3].title) {
+        navigate(`${selectNavLink}/${encodedTitle}`);
+      }
+
+      // const searchParams = new URLSearchParams(location.search); // Create a new instance of URLSearchParams
+
+      // Create an array of navigation links with their corresponding query parameters
+      const navLinkMap = [
```
Comment (Vishnu9901) [inline src/libs/ui/src/organisms/Header/index.tsx]: remove this variable here

File: src/libs/ui/src/organisms/Header/index.tsx [hunk 7]
```diff
@@ -108,32 +109,93 @@ const Header: React.FC<headerProps> = ({ modalSetToggle, handleRegisterClick })
...
+      const selectedLink = navLinkMap.find(link => link.title === selectNavLink);
+
+      if (selectedLink) {
+        console.log(selectedLink.category)
+        if (selectedLink.category === 'Best-Seller' || selectedLink.category === 'View-All') {
+          // Add the selected category to the params
+          categoryParams.push(`category=${selectedLink.category}`);
+
+          // If you want to add more predefined categories, do it here
+          const additionalCategories = [
```
Comment (Vishnu9901) [inline src/libs/ui/src/organisms/Header/index.tsx]: remove this variable from here

File: src/libs/ui/src/organisms/Header/index.tsx [hunk 8]
```diff
@@ -108,32 +109,93 @@ const Header: React.FC<headerProps> = ({ modalSetToggle, handleRegisterClick })
...
+
+          // Add additional categories to the categoryParams
+          additionalCategories.forEach(cat => {
+            categoryParams.push(`category=${encodeURIComponent(cat)}`);
+          });
+
+          // Combine all category params into a single string
+          const finalUrl = `${baseUrl}${categoryParams.join('&')}&best-seller=best-seller`;
+          navigate(finalUrl);
+          window.location.reload(); // Optional: reload the page if needed
```
Comment (Vishnu9901) [inline src/libs/ui/src/organisms/Header/index.tsx]: why this reload needed

File: src/libs/ui/src/organisms/Header/index.tsx [hunk 9]
```diff
@@ -108,32 +109,93 @@ const Header: React.FC<headerProps> = ({ modalSetToggle, handleRegisterClick })
...
+        }
+        else {
+          navigate(`products?category=${selectedLink.category}`);
+          console.log('hello')
+          window.location.reload();
```
Comment (Vishnu9901) [inline src/libs/ui/src/organisms/Header/index.tsx]: please remove this

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+  const [enableBestSeller,setEnableBestSeller]=useState<boolean>(false);
+  // Update URL params when a category is selected
+  const handleCategorySelect = (category: string | null) => {
+    const newCategory = category ?? 'all products';
+
+    // Check if the selected category is the same as the current one
+    if (newCategory === selectedProductCategory) {
+    
+      console.log(selectedProductCategory);
+    }
```
Comment (Vishnu9901) [inline src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: remove unused code

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+    if (newCategory === 'best seller') {
+      setIsBestSellerState(true);
+      setFilters([]);
+      navigate('products?category=best-seller');
+      setSelectedProductCategory(newCategory); // Set the selected category
+      return;
+    }
+
+    // If "All Products" is selected, display all category badges
+    if (newCategory === 'All Products' || newCategory === 'View All') {
```
Comment (Vishnu9901) [inline src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: this one also

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+
+    // For other specific categories
+  
+    const searchParams = new URLSearchParams(location.search);
+  
+    // Only update the category in the URL if it's different from the currently selected one
+    if (newCategory !== 'all products') {
+      searchParams.set('category', newCategory);
+    } 
+
```
Comment (Vishnu9901) [inline src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: create some reusable functions for above logic

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+    // Update the URL with the selected sort option
+    const searchParams = new URLSearchParams(location.search);
+    searchParams.set('sort', option);
+    navigate({ search: searchParams.toString() });
+  };
+
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any 
+  const sortProducts = (products: any[]) => {
+    switch (selectedSortOption) {
+    case 'Alphabetical A - Z':
```
Comment (Vishnu9901) [inline src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: use enum or constants for swith case strings

**Review by Vishnu9901 — CHANGES_REQUESTED** [review]

File: src/libs/ui/src/molecules/PlpAccordians/index.tsx (hunk 1 above)
Comment (swathiiN) [reply src/libs/ui/src/molecules/PlpAccordians/index.tsx]: fixed

File: src/libs/ui/src/molecules/PlpAccordians/index.tsx (hunk 2 above)
Comment (swathiiN) [reply src/libs/ui/src/molecules/PlpAccordians/index.tsx]: removed and added top

File: src/libs/ui/src/molecules/PlpAccordians/index.tsx (hunk 3 above)
Comment (swathiiN) [reply src/libs/ui/src/molecules/PlpAccordians/index.tsx]: fixed

File: src/libs/ui/src/organisms/Header/index.tsx (hunk 8 above)
Comment (swathiiN) [reply src/libs/ui/src/organisms/Header/index.tsx]: removed

File: src/libs/ui/src/organisms/Header/index.tsx (hunk 9 above)
Comment (swathiiN) [reply src/libs/ui/src/organisms/Header/index.tsx]: removed

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx (hunk 10 above)
Comment (swathiiN) [reply src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: remvoed

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx (hunk 13 above)
Comment (swathiiN) [reply src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: created constants for this

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+import { toggleLoginModel } from '@store/services/Slices/ModalSlice';
+import { useNavigate, useLocation } from 'react-router-dom';
+import { plpConstant, plpFilters } from '@utils/plpFilterData';
+
+const PlpPageTemplate: React.FC = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const location = useLocation();
+  // Define sorting options as constants (you can also create this in a separate file)
+  const SortOptions = {
```
Comment (Vishnu9901) [inline src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: remove from here and add in constant file

**Review by Vishnu9901 — CHANGES_REQUESTED** [review]

File: src/libs/ui/src/organisms/Header/index.tsx [hunk 15]
```diff
@@ -117,138 +110,114 @@ const Header: React.FC<headerProps> = ({
...
 
   const handleNavLinkClick = (title: string) => {
-    const catagory = appSetting.find((link) => link.title === selectNavLink);
-    const isNavigate =
-      catagory &&
-      catagory?.canNavigate &&
-      catagory?.navigationPages.includes(title);
-    // const selectedLink = appSetting.find(link => link.title === selectNavLink)?.canNavigate;
-
+    console.log('title', title, selectNavLink)
```
Comment (Vishnu9901) [inline src/libs/ui/src/organisms/Header/index.tsx]: remove consoles

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+  const sortedProducts = sortProducts([...filteredProducts]);
+
+  // Sync category and sort option with URL params
+  useEffect(() => {
+    const searchParams = new URLSearchParams(location.search);
+    const categoryFromUrl = searchParams.get('category');
+    const sortFromUrl = searchParams.get('sort');
+
+    // Check for Best Seller in URL
+    if (categoryFromUrl === plpConstant.bestSeller || categoryFromUrl === plpConstant.viewAll) {
```
Comment (Vishnu9901) [inline src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: plpConstant is should be like PLP_CONSTANTs

**Review by Vishnu9901 — CHANGES_REQUESTED** [review]

File: src/libs/ui/src/organisms/Header/index.tsx [hunk 17]
```diff
@@ -117,138 +110,97 @@ const Header: React.FC<headerProps> = ({
...
+                    If you have a professional account, please login. If you would like to establish a professional account please click Create Account.
+                  </h2>
+                  <div className='pop_up p-4 flex gap-4'>
+                    {/* <PrimaryButton className='w-[152px] font-bold text-base  text-sm font-HeroNewBold'>Login In</PrimaryButton> */}
+                    <CreateAccountButton className='w-[90%] font-HeroNewBold text-sm' onClick={modalSetToggle}>Login</CreateAccountButton>
+                    {/* {toggle && <LoginModal closeModal={modalSetToggle} />} */}
+                    <CreateAccountButton className='w-[90%] font-HeroNewBold text-sm' onClick={handleRegisterClick}>Create Account</CreateAccountButton>
+
+                  </div>
+                </Popover>
```
Comment (Vishnu9901) [inline src/libs/ui/src/organisms/Header/index.tsx]: why you remove above login condition ?, we need it

**Review by Vishnu9901 — CHANGES_REQUESTED** [review]

File: src/libs/ui/src/templates/ProductListPageTemplate/index.tsx [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+    case SortOptions.PRICE_LOW_HIGH:
+      return products.sort((a, b) => a.price - b.price);
+    case SortOptions.PRICE_HIGH_LOW:
+      return products.sort((a, b) => b.price - a.price);
+    default:
+      return products;
+    }
+  };
+
+  const sortedProducts = sortProducts([...filteredProducts]);
```
Comment (Vishnu9901) [inline src/libs/ui/src/templates/ProductListPageTemplate/index.tsx]: use useMemo

**Review by Vishnu9901 — CHANGES_REQUESTED** [review]

**Review by Vishnu9901 — APPROVED** [review]

**Pull request merged**